core/templates/services/prevent-page-unload-event.service.ts

```diff
@@ -25,8 +25,9 @@ import { WindowRef } from 'services/contextual/window-ref.service';
 })
 export class PreventPageUnloadEventService {
   private listenerActive: boolean;
-  validationCallback: Function;
-  _preventPageUnloadEventHandlerBind = undefined;
+  validationCallback?: (() => boolean);
```

Comment:
doesn't `?` serve the same purpose? It adds `| undefined` to the type

Reply:
I think using `?` is making the property optional and making it `| undefined` means that we can assign a value of undefined to that particular variable.

Reply:
when we make a property optional, it is actually implicitly being made of union type with `| undefined`. but I have explicitly updated the type with `| undefined` since we are assigning it an `undefined` value.